qiime2/sdk/result.py

```diff
@@ -480,6 +530,144 @@ def merge_annotations(self, other):
                     else:
                         raise e
 
+    def verify(self, signature_name):
+        """
+        Verify a Signature annotation by name on the provided Result.
+
+        Parameters
+        ----------
+        signature_name
+            Name of the Signature Annotation to verify.
+
+        Returns
+        -------
+        dict
+            {
+            'ok': bool,
+            'fingerprint_ok': bool,
+            'checksum_digest_ok': bool,
+            'gpg_signature_ok': bool,
+            'file_checksums_ok': bool,
+            'missing_files': [relpath, ...],
+            'mismatched_files':
+                [{'path': relpath, 'expected': hex, 'actual': hex}, ...],
+            'details': str
+            'fingerprint': str | None,
+            'signer_name': str | None,
+            'signer_email': str | None
+            }
+
+        Notes
+        -----
+        The following checks are performed:
+            - keypair presence / fingerprint match in local GPG keyring
+            - sha512sum for root level checksums file matches checksum_digest
+            - gpg detached signature verification
+            - sha512sum checks for each file in signature-level checksums file
+        """
+        # first make sure the Result isn't malformed & the Signature exists
+        self.validate()
+        signature = self.get_annotation(signature_name)
+
+        report = self._empty_report(signature)
+
+        annotation_dir = \
+            pathlib.Path(self._archiver.annotations_dir) / str(signature.id)
+
+        root_fp = self._archiver.root_dir
+        root_checksums_fp = root_fp / 'checksums.sha512'
+        sig_checksums_fp = annotation_dir / 'checksums.sha512'
+        signature_fp = annotation_dir / 'signature.gpg'
+
+        try:
+            keypair_selector = (getattr(signature, 'fingerprint') or
+                                getattr(signature, 'signer_uid')
+                                )
+            if not keypair_selector:
+                report['details'] = 'Signature is missing fingerprint/UID.'
+                return report
+            found_keypair = gpg_find_key(keypair_selector)
+            if getattr(signature, 'fingerprint'):
+                report['fingerprint_ok'] = (
+                    normalize_fingerprint(found_keypair['fingerprint']) ==
+                    normalize_fingerprint(signature.fingerprint)
+                )
```

Comment:
`signer_uid` doesn't exist anymore

```suggestion
            fingerprint = getattr(signature, 'fingerprint') 
            if not fingerprint:
                report['details'] = 'Signature is missing fingerprint.'
                return report # Some other form of control flow that doesn't dodge the error
                
             found_keypair = gpg_find_key(fingerprint)               
             report['fingerprint_ok'] = (
                    normalize_fingerprint(found_keypair['fingerprint']) ==
                    normalize_fingerprint(fingerprint)
                )
```